src/js/gi/components/Dropdown.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+class Dropdown extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: props.defaultOption
+    };
+  }
+
+  // renders a list of options defined as an array
+  // in the form of [value1, label1, value2, label2...]
+  renderDropdownOptions(optionsArray) {
+    const options = Object.assign([], optionsArray);
```

Comment:
I don't think `Object.assign` is intended for cloning arrays. Instead, maybe:

`const options = optionsArray.slice();`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice